lib/spack/spack/test/architecture.py

```diff
@@ -58,7 +58,7 @@ def test_dict_functions_for_architecture():
 
 def test_platform():
         output_platform_class = spack.architecture.real_platform()
-        if os.path.exists('/opt/cray/craype'):
+        if os.environ.get('CRAYPE_VERSION') is not None:
```

Comment:
In case other reviewers are curious, the new way is how we already detect the cray platform in platforms/cray.py as well.